### ISSUE TITLE
Please add option to allow migration to new URL

### ISSUE THREAD

**Issue by sirzooro** [BOINC/boinc#2111]
Projects sometimes have to change their URL, usually when they adds SSL certificate. There are also cases when project starts with IP address and then gets domain name. Now official way to handle this (from client perspective) is to disconnect from project and connect again using new URL. This also requires to finish all downloaded WUs and report them back to server, otherwise any progress will be lost. Please add new option to client which will allow to switch project to new URL, so all downloaded WUs and work progress would be preserved.

It would be also nice to have some redirect support on server side, so whole process could happen with no or little user interaction - some kind of confirmation from user probably should be there for security purposes.

**Comment by ChristianBeer** [comment]
The problem with changing the URL of a project is security. How can we make the mechanism easy enough to be usable yet secure enough so it can not be exploited.

This mechanism can be used to divert Clients from an existing project to a fake project that looks like the real project. The big question is how would the client verify that the request to change the project URL is legitimate and was not send by an attacker (e.g. via a man-in-the-middle attack)?

**Comment by AenBleidd** [comment]
@ChristianBeer,
I guess we should add such functionality ant let user to decide whether he
or she wants to automatically change the URL from [old] to [new] or not.
Specially for me this is very useful functionality because I do not want to
wait a moment when all current tasks will be finished to manually remove
and re-add the project. It is very painful when you have several devices
and you need to remember where you made such change and which devices are
pending.
I think if we will let user to decide whether he or she wants to change the
URL of the particular project - it is enough from the security POV.

Thanks

Best regards,
Vitalii Koshura

2017-09-12 17:40 GMT+03:00 Christian Beer <notifications@github.com>:

**Comment by sirzooro** [comment]
Security could be increased by requesting some token from new URL and sending it to old URL for verification. This should be discussed with some security expert how to do this properly.

BTW, Case of migration from http to https is simpler, because server address does not change. Server still can be configured to serve different sites for http and https, but for this user must be able to login via SSH to some account with appropriate privileges. When someone will do this, he will be able to to do much more bad things, so this case is of less importance here.

**Comment by sibebleuze** [comment]
Also, when you want to change url and add the new one first, the 'delete project' button doesn't work on the url that you want to delete.

**Comment by AenBleidd** [comment]
This is not something that happens very often. And to be completely honest, Projects should not change their URL. If the URL is changed - it's a very bad sign, and having such an option available for users is very risky.
Currently there is already a mechanism to migrate silently from HTTP to HTTPS URL, and this doesn't require any user interaction.
Moving from IP address to the real URL - is something that also happens only once before the project goes public, and thus I don't believe we need to cover such a case from the client.

Thus, closing this for now.
